src/components/Filter/SearchFilter.jsx

```diff
@@ -0,0 +1,58 @@
+import { useDataContext, videoConstants } from "../../context";
+
+import { FiSearch } from "react-icons/fi";
+import "./search-filter.scss";
+
+const SearchFilter = ({ categories }) => {
+  const {
+    dataState: {
+      vid: {
+        filterOptions: { category },
+      },
+    },
+    dataDispatch,
+  } = useDataContext();
+
+  const handleSearch = (e) => {
+    setTimeout(() => {
+      dataDispatch({
+        type: videoConstants.SEARCH,
+        payload: e.target.value,
+      });
+    }, 1000);
+  };
+
+  return (
+    <div className="search-filter b-margin-lg">
+      <select
+        className="category-dropdown"
+        value={category}
+        onChange={(e) =>
+          dataDispatch({
+            type: videoConstants.FILTER_CATEGORY,
+            payload: e.target.value,
+          })
+        }
```

Comment:
You can extract this function out for easier debugging

Reply:
Fixed - cf24a737c24c76d8caf3e44ab97cbcf0b2e0c37b